share/scan/registry.go

```diff
@@ -70,52 +70,79 @@ func IsQuayRegistry(rc *RegClient) bool {
 	return strings.EqualFold(rc.URL[:len(quayRegistryURL)], quayRegistryURL)
 }
 
-func BuildV2ImageInfo(dg string, body []byte) (imageInfo ImageInfo, parsedSchemaVersion int, err error) {
+func (rc *RegClient) copyV2LayerAndHistory(imageInfo *ImageInfo, manV2 *manifestV2.Manifest, ccmi *registry.ManifestInfo) bool {
+	allLayersAreCosignPayloads := true
+
```

Comment:
Should imageInfo.Layers be reset to empty first after entering this function?
Or imageInfo.Layers is always empty when copyV2LayerAndHistory() is called?

Reply:
It is initiated in GetImageInfo(). It is always empty when entering here.

Reply:
Maybe we should, because there are two places where buildV2ImageInfo() is called.